src/Repository/Provider/InMemoryRepositoryProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Repository\Provider;
```

Comment:
I think this is somehow generic so could be in the root namespace. But you're right, that's not coherent with the rest of the code.